Завдання 1. Завантажити датасет за посиланням https://www.kaggle.com/datasets/aravindpcoder/obesity-or-cvd-risk-classifyregressorcluster

Датасет https://drive.google.com/file/d/1lWbCGgstFXi2hiTRSwFt1Tph_tjfMsyY/view?usp=drive_link

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, accuracy_score

In [ ]:
from google.colab import files

# Завантаження файлу
uploaded = files.upload()

Saving ObesityDataSet.csv to ObesityDataSet.csv


In [ ]:
# Завантаження CSV-файлу
df = pd.read_csv('ObesityDataSet.csv')

# Перегляд перших 5 рядків
df

Gender        Age    Height      Weight family_history_with_overweight  \
0     Female  21.000000  1.620000   64.000000                            yes   
1     Female  21.000000  1.520000   56.000000                            yes   
2       Male  23.000000  1.800000   77.000000                            yes   
3       Male  27.000000  1.800000   87.000000                             no   
4       Male  22.000000  1.780000   89.800000                             no   
...      ...        ...       ...         ...                            ...   
2106  Female  20.976842  1.710730  131.408528                            yes   
2107  Female  21.982942  1.748584  133.742943                            yes   
2108  Female  22.524036  1.752206  133.689352                            yes   
2109  Female  24.361936  1.739450  133.346641                            yes   
2110  Female  23.664709  1.738836  133.472641                            yes   

     FAVC  FCVC  NCP       CAEC SMOKE      CH2O  SCC       FAF       TUE  \
0      no   2.0  3.0  Sometimes    no  2.000000   no  0.000000  1.000000   
1      no   3.0  3.0  Sometimes   yes  3.000000  yes  3.000000  0.000000   
2      no   2.0  3.0  Sometimes    no  2.000000   no  2.000000  1.000000   
3      no   3.0  3.0  Sometimes    no  2.000000   no  2.000000  0.000000   
4      no   2.0  1.0  Sometimes    no  2.000000   no  0.000000  0.000000   
...   ...   ...  ...        ...   ...       ...  ...       ...       ...   
2106  yes   3.0  3.0  Sometimes    no  1.728139   no  1.676269  0.906247   
2107  yes   3.0  3.0  Sometimes    no  2.005130   no  1.341390  0.599270   
2108  yes   3.0  3.0  Sometimes    no  2.054193   no  1.414209  0.646288   
2109  yes   3.0  3.0  Sometimes    no  2.852339   no  1.139107  0.586035   
2110  yes   3.0  3.0  Sometimes    no  2.863513   no  1.026452  0.714137   

            CALC                 MTRANS           NObeyesdad  
0             no  Public_Transportation        Normal_Weight  
1      Sometimes  Public_Transportation        Normal_Weight  
2     Frequently  Public_Transportation        Normal_Weight  
3     Frequently                Walking   Overweight_Level_I  
4      Sometimes  Public_Transportation  Overweight_Level_II  
...          ...                    ...                  ...  
2106   Sometimes  Public_Transportation     Obesity_Type_III  
2107   Sometimes  Public_Transportation     Obesity_Type_III  
2108   Sometimes  Public_Transportation     Obesity_Type_III  
2109   Sometimes  Public_Transportation     Obesity_Type_III  
2110   Sometimes  Public_Transportation     Obesity_Type_III  

[2111 rows x 17 columns]

Дані складаються з оцінки рівня ожиріння у людей із Мексики, Перу та Колумбії віком від 14 до 61 року та різноманітними харчовими звичками та фізичним станом. Дані були зібрані за допомогою веб-платформи з опитуванням, у якому анонімні користувачі відповідали на кожне запитання. питання, потім інформація була оброблена, отримано 17 атрибутів і 2111 записів.
1.**Gender** – стать (Male – чоловік, Female – жінка).

2. **Age** – вік (у роках).

3. **Height** – зріст (у метрах).

4. **Weight** – вага (у кілограмах).

5. **family_history_with_overweight** – наявність сімейної історії зайвої ваги (yes – є, no – немає).
Це свідчить про генетичну схильність.

6. **FAVC** (Frequent consumption of high caloric food) – часте вживання висококалорійної їжі (yes/no).

7. **FCVC** (Frequency of consumption of vegetables) – частота споживання овочів (зазвичай шкала від 1 до 3, де 1 – рідко, 3 – часто).

8. **NCP** (Number of main meals per day) – кількість основних прийомів їжі на день.

9. **CAEC** (Consumption of food between meals) – вживання їжі між основними прийомами (No, Sometimes, Frequently, Always).

10. **SMOKE** – куріння (yes/no).

11. **CH2O** (Consumption of water per day) – споживання води на день (зазвичай у шкалі, наприклад, від 1 до 3).

12. **SCC** (Calories monitoring – Self control of calories consumption) – чи контролює людина кількість спожитих калорій (yes/no).

13.**FAF** (Physical activity frequency) – частота фізичної активності (у годинах на тиждень).

14. **TUE** (Time using technology devices – TV, computer, phone) – кількість годин на день, проведених із ґаджетами.

15. **CALC** (Consumption of alcohol) – споживання алкоголю (No, Sometimes, Frequently, Always).

16. **MTRANS** (Transportation method) – основний спосіб пересування (Automobile, Motorbike, Bike, Public_Transportation, Walking).

17. **NObeyesdad** – цільова змінна: класифікація стану ваги (наприклад:


Значення NOwesity є:

1. Недостатня вага Менше 18,5
2. Нормальна 18,5 до 24,9
3. Надмірна вага 25,0 до 29,9
4. Ожиріння I 30,0 до 34,9
5. Ожиріння II 35,0 до 39,9
6.  Ожиріння III Більше 40

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Gender                          2111 non-null   object 
 1   Age                             2111 non-null   float64
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   family_history_with_overweight  2111 non-null   object 
 5   FAVC                            2111 non-null   object 
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   CAEC                            2111 non-null   object 
 9   SMOKE                           2111 non-null   object 
 10  CH2O                            2111 non-null   float64
 11  SCC                             2111 non-null   object 
 12  FAF                             21

In [ ]:
df.describe()

Age       Height       Weight         FCVC          NCP  \
count  2111.000000  2111.000000  2111.000000  2111.000000  2111.000000   
mean     24.312600     1.701677    86.586058     2.419043     2.685628   
std       6.345968     0.093305    26.191172     0.533927     0.778039   
min      14.000000     1.450000    39.000000     1.000000     1.000000   
25%      19.947192     1.630000    65.473343     2.000000     2.658738   
50%      22.777890     1.700499    83.000000     2.385502     3.000000   
75%      26.000000     1.768464   107.430682     3.000000     3.000000   
max      61.000000     1.980000   173.000000     3.000000     4.000000   

              CH2O          FAF          TUE  
count  2111.000000  2111.000000  2111.000000  
mean      2.008011     1.010298     0.657866  
std       0.612953     0.850592     0.608927  
min       1.000000     0.000000     0.000000  
25%       1.584812     0.124505     0.000000  
50%       2.000000     1.000000     0.625350  
75%       2.477420     1.666678     1.000000  
max       3.000000     3.000000     2.000000

In [ ]:
df['Gender'].value_counts()

Gender
Male      1068
Female    1043
Name: count, dtype: int64

In [ ]:
df['MTRANS'].value_counts()

MTRANS
Public_Transportation    1580
Automobile                457
Walking                    56
Motorbike                  11
Bike                        7
Name: count, dtype: int64

In [ ]:
df['family_history_with_overweight'].value_counts()

family_history_with_overweight
yes    1726
no      385
Name: count, dtype: int64

In [ ]:
df['CAEC'].value_counts()

CAEC
Sometimes     1765
Frequently     242
Always          53
no              51
Name: count, dtype: int64

In [ ]:
df['FAVC'].value_counts()

FAVC
yes    1866
no      245
Name: count, dtype: int64

In [ ]:
df['SCC'].value_counts()

SCC
no     2015
yes      96
Name: count, dtype: int64

In [ ]:
df['CALC'].value_counts()

CALC
Sometimes     1401
no             639
Frequently      70
Always           1
Name: count, dtype: int64

In [ ]:
df['MTRANS'].value_counts()

MTRANS
Public_Transportation    1580
Automobile                457
Walking                    56
Motorbike                  11
Bike                        7
Name: count, dtype: int64

In [ ]:
df['NObeyesdad'].value_counts()

NObeyesdad
Obesity_Type_I         351
Obesity_Type_III       324
Obesity_Type_II        297
Overweight_Level_I     290
Overweight_Level_II    290
Normal_Weight          287
Insufficient_Weight    272
Name: count, dtype: int64

In [ ]:
df

Gender        Age    Height      Weight family_history_with_overweight  \
0     Female  21.000000  1.620000   64.000000                            yes   
1     Female  21.000000  1.520000   56.000000                            yes   
2       Male  23.000000  1.800000   77.000000                            yes   
3       Male  27.000000  1.800000   87.000000                             no   
4       Male  22.000000  1.780000   89.800000                             no   
...      ...        ...       ...         ...                            ...   
2106  Female  20.976842  1.710730  131.408528                            yes   
2107  Female  21.982942  1.748584  133.742943                            yes   
2108  Female  22.524036  1.752206  133.689352                            yes   
2109  Female  24.361936  1.739450  133.346641                            yes   
2110  Female  23.664709  1.738836  133.472641                            yes   

     FAVC  FCVC  NCP       CAEC SMOKE      CH2O  SCC       FAF       TUE  \
0      no   2.0  3.0  Sometimes    no  2.000000   no  0.000000  1.000000   
1      no   3.0  3.0  Sometimes   yes  3.000000  yes  3.000000  0.000000   
2      no   2.0  3.0  Sometimes    no  2.000000   no  2.000000  1.000000   
3      no   3.0  3.0  Sometimes    no  2.000000   no  2.000000  0.000000   
4      no   2.0  1.0  Sometimes    no  2.000000   no  0.000000  0.000000   
...   ...   ...  ...        ...   ...       ...  ...       ...       ...   
2106  yes   3.0  3.0  Sometimes    no  1.728139   no  1.676269  0.906247   
2107  yes   3.0  3.0  Sometimes    no  2.005130   no  1.341390  0.599270   
2108  yes   3.0  3.0  Sometimes    no  2.054193   no  1.414209  0.646288   
2109  yes   3.0  3.0  Sometimes    no  2.852339   no  1.139107  0.586035   
2110  yes   3.0  3.0  Sometimes    no  2.863513   no  1.026452  0.714137   

            CALC                 MTRANS           NObeyesdad  
0             no  Public_Transportation        Normal_Weight  
1      Sometimes  Public_Transportation        Normal_Weight  
2     Frequently  Public_Transportation        Normal_Weight  
3     Frequently                Walking   Overweight_Level_I  
4      Sometimes  Public_Transportation  Overweight_Level_II  
...          ...                    ...                  ...  
2106   Sometimes  Public_Transportation     Obesity_Type_III  
2107   Sometimes  Public_Transportation     Obesity_Type_III  
2108   Sometimes  Public_Transportation     Obesity_Type_III  
2109   Sometimes  Public_Transportation     Obesity_Type_III  
2110   Sometimes  Public_Transportation     Obesity_Type_III  

[2111 rows x 17 columns]

In [ ]:
df.duplicated().sum()

np.int64(24)

In [ ]:
df = df.drop_duplicates()
df

Gender        Age    Height      Weight family_history_with_overweight  \
0     Female  21.000000  1.620000   64.000000                            yes   
1     Female  21.000000  1.520000   56.000000                            yes   
2       Male  23.000000  1.800000   77.000000                            yes   
3       Male  27.000000  1.800000   87.000000                             no   
4       Male  22.000000  1.780000   89.800000                             no   
...      ...        ...       ...         ...                            ...   
2106  Female  20.976842  1.710730  131.408528                            yes   
2107  Female  21.982942  1.748584  133.742943                            yes   
2108  Female  22.524036  1.752206  133.689352                            yes   
2109  Female  24.361936  1.739450  133.346641                            yes   
2110  Female  23.664709  1.738836  133.472641                            yes   

     FAVC  FCVC  NCP       CAEC SMOKE      CH2O  SCC       FAF       TUE  \
0      no   2.0  3.0  Sometimes    no  2.000000   no  0.000000  1.000000   
1      no   3.0  3.0  Sometimes   yes  3.000000  yes  3.000000  0.000000   
2      no   2.0  3.0  Sometimes    no  2.000000   no  2.000000  1.000000   
3      no   3.0  3.0  Sometimes    no  2.000000   no  2.000000  0.000000   
4      no   2.0  1.0  Sometimes    no  2.000000   no  0.000000  0.000000   
...   ...   ...  ...        ...   ...       ...  ...       ...       ...   
2106  yes   3.0  3.0  Sometimes    no  1.728139   no  1.676269  0.906247   
2107  yes   3.0  3.0  Sometimes    no  2.005130   no  1.341390  0.599270   
2108  yes   3.0  3.0  Sometimes    no  2.054193   no  1.414209  0.646288   
2109  yes   3.0  3.0  Sometimes    no  2.852339   no  1.139107  0.586035   
2110  yes   3.0  3.0  Sometimes    no  2.863513   no  1.026452  0.714137   

            CALC                 MTRANS           NObeyesdad  
0             no  Public_Transportation        Normal_Weight  
1      Sometimes  Public_Transportation        Normal_Weight  
2     Frequently  Public_Transportation        Normal_Weight  
3     Frequently                Walking   Overweight_Level_I  
4      Sometimes  Public_Transportation  Overweight_Level_II  
...          ...                    ...                  ...  
2106   Sometimes  Public_Transportation     Obesity_Type_III  
2107   Sometimes  Public_Transportation     Obesity_Type_III  
2108   Sometimes  Public_Transportation     Obesity_Type_III  
2109   Sometimes  Public_Transportation     Obesity_Type_III  
2110   Sometimes  Public_Transportation     Obesity_Type_III  

[2087 rows x 17 columns]

In [ ]:
X = df.drop('NObeyesdad', axis=1)
y = df['NObeyesdad']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [ ]:
# Масштабування та кодування
# Визначення категоріальних і числових колонок
categorical_features = ['Gender', 'family_history_with_overweight', 'FAVC', 'SCC', 'SMOKE', 'MTRANS']
numerical_features = X.select_dtypes(include=['int64', 'float64']).columns.tolist()

# Пайплайн для обробки
preprocessor = ColumnTransformer(
    transformers=[
        ('num', StandardScaler(), numerical_features),  # Масштабування числових ознак
        ('cat', OneHotEncoder(), categorical_features)  # Одноразове кодування категоріальних ознак
    ])

In [ ]:
# Побудова моделі логістичної регресії
log_reg = Pipeline(steps=[('preprocessor', preprocessor),
                           ('classifier', LogisticRegression(max_iter=1000))])
log_reg.fit(X_train, y_train)

Pipeline(steps=[('preprocessor',
                 ColumnTransformer(transformers=[('num', StandardScaler(),
                                                  ['Age', 'Height', 'Weight',
                                                   'FCVC', 'NCP', 'CH2O', 'FAF',
                                                   'TUE']),
                                                 ('cat', OneHotEncoder(),
                                                  ['Gender',
                                                   'family_history_with_overweight',
                                                   'FAVC', 'SCC', 'SMOKE',
                                                   'MTRANS'])])),
                ('classifier', LogisticRegression(max_iter=1000))])

In [ ]:
# Побудова моделі RandomForestClassifier
rf_classifier = Pipeline(steps=[('preprocessor', preprocessor),
                                 ('classifier', RandomForestClassifier(n_estimators=100, random_state=42))])
rf_classifier.fit(X_train, y_train)

Pipeline(steps=[('preprocessor',
                 ColumnTransformer(transformers=[('num', StandardScaler(),
                                                  ['Age', 'Height', 'Weight',
                                                   'FCVC', 'NCP', 'CH2O', 'FAF',
                                                   'TUE']),
                                                 ('cat', OneHotEncoder(),
                                                  ['Gender',
                                                   'family_history_with_overweight',
                                                   'FAVC', 'SCC', 'SMOKE',
                                                   'MTRANS'])])),
                ('classifier', RandomForestClassifier(random_state=42))])

In [ ]:
# Оцінка моделей
y_pred_log_reg = log_reg.predict(X_test)
print("Логістична регресія:")
print(classification_report(y_test, y_pred_log_reg))
print("Точність:", accuracy_score(y_test, y_pred_log_reg))

Логістична регресія:
                     precision    recall  f1-score   support

Insufficient_Weight       0.86      0.95      0.90        59
      Normal_Weight       0.90      0.72      0.80        61
     Obesity_Type_I       0.94      0.93      0.94        70
    Obesity_Type_II       0.97      1.00      0.98        64
   Obesity_Type_III       1.00      1.00      1.00        60
 Overweight_Level_I       0.76      0.76      0.76        55
Overweight_Level_II       0.72      0.80      0.76        49

           accuracy                           0.89       418
          macro avg       0.88      0.88      0.88       418
       weighted avg       0.89      0.89      0.88       418

Точність: 0.8851674641148325


In [ ]:
y_pred_rf = rf_classifier.predict(X_test)
print("\nВипадковий ліс:")
print(classification_report(y_test, y_pred_rf))
print("Точність:", accuracy_score(y_test, y_pred_rf))


Випадковий ліс:
                     precision    recall  f1-score   support

Insufficient_Weight       0.96      0.92      0.94        59
      Normal_Weight       0.82      0.90      0.86        61
     Obesity_Type_I       0.99      0.96      0.97        70
    Obesity_Type_II       1.00      1.00      1.00        64
   Obesity_Type_III       1.00      1.00      1.00        60
 Overweight_Level_I       0.94      0.87      0.91        55
Overweight_Level_II       0.90      0.96      0.93        49

           accuracy                           0.94       418
          macro avg       0.95      0.94      0.94       418
       weighted avg       0.95      0.94      0.95       418

Точність: 0.9449760765550239


Підібрати оптимальні параметри за допомогою GridSearchCV

In [ ]:
# Підбір параметрів для логістичної регресії
#Pipeline — це клас із бібліотеки scikit-learn, що дозволяє послідовно виконувати кілька кроків обробки.

# Сітка параметрів для логістичної регресії
log_reg_param_grid = {
    'classifier__C': [0.1, 1, 10, 100],  # Регуляризація
    'classifier__solver': ['liblinear', 'lbfgs']  # Алгоритми оптимізації
}

log_reg_grid_search = GridSearchCV(log_reg, log_reg_param_grid, cv=5, scoring='accuracy')
log_reg_grid_search.fit(X_train, y_train)

GridSearchCV(cv=5,
             estimator=Pipeline(steps=[('preprocessor',
                                        ColumnTransformer(transformers=[('num',
                                                                         StandardScaler(),
                                                                         ['Age',
                                                                          'Height',
                                                                          'Weight',
                                                                          'family_history_with_overweight',
                                                                          'FCVC',
                                                                          'NCP',
                                                                          'CH2O',
                                                                          'FAF',
                                                                          'TUE']),
                                                                        ('cat',
                                                                         OneHotEncoder(),
                                                                         ['Gender',
                                                                          'family_history_with_overweight',
                                                                          'FAVC',
                                                                          'SCC',
                                                                          'SMOKE',
                                                                          'MTRANS'])])),
                                       ('classifier',
                                        LogisticRegression(max_iter=1000))]),
             param_grid={'classifier__C': [0.1, 1, 10, 100],
                         'classifier__solver': ['liblinear', 'lbfgs']},
             scoring='accuracy')

In [ ]:
# Підбір параметрів для випадкового лісу

# Сітка параметрів для випадкового лісу
rf_param_grid = {
    'classifier__n_estimators': [100, 200, 300],  # Кількість дерев
    'classifier__max_depth': [None, 10, 20, 30],  # Максимальна глибина дерева
    'classifier__min_samples_split': [2, 5, 10],  # Мінімальна кількість зразків для розбиття вузла
    'classifier__min_samples_leaf': [1, 2, 4]  # Мінімальна кількість зразків у листку
}

rf_grid_search = GridSearchCV(rf_classifier, rf_param_grid, cv=5, scoring='accuracy')
rf_grid_search.fit(X_train, y_train)

GridSearchCV(cv=5,
             estimator=Pipeline(steps=[('preprocessor',
                                        ColumnTransformer(transformers=[('num',
                                                                         StandardScaler(),
                                                                         ['Age',
                                                                          'Height',
                                                                          'Weight',
                                                                          'family_history_with_overweight',
                                                                          'FCVC',
                                                                          'NCP',
                                                                          'CH2O',
                                                                          'FAF',
                                                                          'TUE']),
                                                                        ('cat',
                                                                         OneHotEncoder(),
                                                                         ['Gender',
                                                                          'family_history_with_overweight',
                                                                          'FAVC',
                                                                          'SCC',
                                                                          'SMOKE',
                                                                          'MTRANS'])])),
                                       ('classifier',
                                        RandomForestClassifier(random_state=42))]),
             param_grid={'classifier__max_depth': [None, 10, 20, 30],
                         'classifier__min_samples_leaf': [1, 2, 4],
                         'classifier__min_samples_split': [2, 5, 10],
                         'classifier__n_estimators': [100, 200, 300]},
             scoring='accuracy')

In [ ]:
# Оцінка результатів
# Логістична регресія з найкращими параметрами
best_log_reg = log_reg_grid_search.best_estimator_
y_pred_log_reg = best_log_reg.predict(X_test)
print("Логістична регресія з найкращими параметрами:")
print("Найкращі параметри:", log_reg_grid_search.best_params_)
print(classification_report(y_test, y_pred_log_reg))
print("Точність:", accuracy_score(y_test, y_pred_log_reg))

Логістична регресія з найкращими параметрами:
Найкращі параметри: {'classifier__C': 100, 'classifier__solver': 'lbfgs'}
                     precision    recall  f1-score   support

Insufficient_Weight       0.94      0.98      0.96        59
      Normal_Weight       0.98      0.93      0.96        61
     Obesity_Type_I       1.00      0.97      0.99        70
    Obesity_Type_II       0.97      1.00      0.98        64
   Obesity_Type_III       1.00      1.00      1.00        60
 Overweight_Level_I       0.94      0.89      0.92        55
Overweight_Level_II       0.88      0.94      0.91        49

           accuracy                           0.96       418
          macro avg       0.96      0.96      0.96       418
       weighted avg       0.96      0.96      0.96       418

Точність: 0.9617224880382775


In [ ]:
# Випадковий ліс з найкращими параметрами
best_rf = rf_grid_search.best_estimator_
y_pred_rf = best_rf.predict(X_test)
print("\nВипадковий ліс з найкращими параметрами:")
print("Найкращі параметри:", rf_grid_search.best_params_)
print(classification_report(y_test, y_pred_rf))
print("Точність:", accuracy_score(y_test, y_pred_rf))


Випадковий ліс з найкращими параметрами:
Найкращі параметри: {'classifier__max_depth': 20, 'classifier__min_samples_leaf': 1, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}
                     precision    recall  f1-score   support

Insufficient_Weight       0.95      0.92      0.93        59
      Normal_Weight       0.80      0.92      0.85        61
     Obesity_Type_I       1.00      0.94      0.97        70
    Obesity_Type_II       1.00      1.00      1.00        64
   Obesity_Type_III       1.00      1.00      1.00        60
 Overweight_Level_I       0.98      0.85      0.91        55
Overweight_Level_II       0.91      0.98      0.94        49

           accuracy                           0.94       418
          macro avg       0.95      0.94      0.94       418
       weighted avg       0.95      0.94      0.95       418

Точність: 0.9449760765550239


In [ ]:
# Отримання 10 справжніх значень з тестового набору
y_true = y_test[:10].values

# Прогнозування 10 значень за допомогою найкращої моделі логістичної регресії
y_pred_log_reg = best_log_reg.predict(X_test[:10])

# Виведення номера пацієнта, справжніх і прогнозованих значень
print("Номер пацієнта | Справжнє значення | Прогнозоване значення")
for i, index in enumerate(y_test[:10].index):
    print(f"Пацієнт {index} | Справжнє: {y_true[i]} | Прогнозоване: {y_pred_log_reg[i]}")


Номер пацієнта | Справжнє значення | Прогнозоване значення
Пацієнт 1337 | Справжнє: Obesity_Type_I | Прогнозоване: Obesity_Type_I
Пацієнт 1763 | Справжнє: Obesity_Type_II | Прогнозоване: Obesity_Type_II
Пацієнт 389 | Справжнє: Overweight_Level_I | Прогнозоване: Overweight_Level_I
Пацієнт 595 | Справжнє: Insufficient_Weight | Прогнозоване: Insufficient_Weight
Пацієнт 1735 | Справжнє: Obesity_Type_II | Прогнозоване: Obesity_Type_II
Пацієнт 900 | Справжнє: Overweight_Level_I | Прогнозоване: Overweight_Level_I
Пацієнт 243 | Справжнє: Overweight_Level_II | Прогнозоване: Overweight_Level_II
Пацієнт 1847 | Справжнє: Obesity_Type_III | Прогнозоване: Obesity_Type_III
Пацієнт 624 | Справжнє: Insufficient_Weight | Прогнозоване: Insufficient_Weight
Пацієнт 51 | Справжнє: Normal_Weight | Прогнозоване: Normal_Weight
